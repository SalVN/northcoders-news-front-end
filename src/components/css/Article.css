.article-title {
    color: black !important;
    font-weight: bold !important;
}

.article-subheading-key {
    margin-right: 5px;
}

.article-subheading-value {
    margin-right: 10px; 
    color: #c1002e;
}
/*.delete-button {
    background-color: #c1002e;
    margin-left: 30px;
    margin-top: 10px;
    color: white;    
}

.article-box-style {
    margin: 10px;
}

.article-number {
    font-size: 20;
    font-weight: bold;
    color: #c1002e;
}


.article-card-subheading-value-notlink {
    margin-right: 10px; 
    color: #000;
}

.article-card-avatar {
    height: 25px;
    margin-right: 5px;
    margin-bottom: -7px;
    border-radius: 20px;
}*/