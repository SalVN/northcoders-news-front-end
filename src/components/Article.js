import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import './css/Article.css';

const Article = function (props) {
    return (
        <div className='box'>
            <div className='columns'>
                <div className='column is-1'>
                    <button className='button is-black is-inverted is-small'><i className="fa fa-arrow-up" aria-hidden="true"></i></button>
                    <p className='comment-card-votes'><strong>{props.article.votes}</strong></p>
                    <button className='button is-black is-inverted is-small'><i className="fa fa-arrow-down" aria-hidden="true"></i></button>
                </div>
                <div className='column'>
                    <div className='title is-3 article-title'>{props.article.title}</div>
                    <p>
                        <span>
                            <span className='article-subheading-key'>added by:</span>
                            <Link to='#'>
                                <span>
                                    {props.user &&
                                        <img className='article-card-avatar' src={props.user.avatar_url} alt='user avatar' />
                                    }
                                </span>
                                <span className='article-subheading-value'>
                                    <strong>{props.article.created_by}</strong>
                                </span>
                            </Link>
                        </span>
                        <span>
                            <span className='article-card-subheading-key'>tags:</span>
                            <Link to={`/topics/${props.article.belongs_to}/articles`}>
                                <span className='article-card-subheading-value'>
                                    <strong>{props.article.belongs_to}</strong>
                                </span>
                            </Link>
                        </span>
                        <span>
                            <span className='article-card-subheading-key'>comments:</span>
                            {/*<Link to='#'>*/}
                            <span className='article-card-subheading-value-notlink'>
                                <strong>{props.article.commentCount}</strong>
                            </span>
                            {/*</Link>*/}
                        </span>
                    </p>
                    <div>{props.article.body}</div>
                </div>
            </div>
        </div>
    );
};

Article.propTypes = {
    article: PropTypes.object.isRequired,
    user: PropTypes.object.isRequired
};

export default Article;