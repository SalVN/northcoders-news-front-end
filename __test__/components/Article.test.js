import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Article from '../../src/components/Article';

describe('Article', () => {
    const article = {
        _id: '59b01acf006c8dbca914672f',
        title: 'Football is fun',
        body: 'something',
        belongs_to: 'football',
        __v: 0,
        votes: 3,
        comment_count: 0
    };
    const user = {
        _id: '59b1b18b327cce1fb043bdb2',
        username: 'northcoder',
        name: 'Awesome Northcoder',
        avatar_url: 'https://avatars3.githubusercontent.com/u/6791502?v=3&s=200',
        __v: 0
    };
    it('is a function', () => {
        expect(typeof Article).toEqual('function');
    });

    it('renders', () => {
        const enzymeWrapper = shallow(<Article
            voteArticle={(x) => { return x; }}
            user={user}
            article={article}
        />);
        expect(enzymeWrapper.children().length).toEqual(1);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <Article
                    voteArticle={(x) => { return x; }}
                    user={user}
                    article={article}
                />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});