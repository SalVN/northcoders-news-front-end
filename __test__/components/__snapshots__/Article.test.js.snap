// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `
<div
  className="box article-card-style"
>
  <div
    className="columns is-mobile"
  >
    <div
      className="column is-1"
    >
      <button
        className="button is-black is-inverted is-small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-arrow-up"
        />
      </button>
      <p
        className="comment-card-votes article-votes"
      >
        <strong>
          3
        </strong>
      </p>
      <button
        className="button is-black is-inverted is-small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-arrow-down"
        />
      </button>
    </div>
    <div
      className="column"
    >
      <div
        className="title is-3 article-title"
      >
        Football is fun
      </div>
      <p>
        <span>
          <span
            className="article-subheading-key"
          >
            added by:
          </span>
          <a
            href="/users/undefined"
            onClick={[Function]}
          >
            <span>
              <img
                alt="user avatar"
                className="article-card-avatar"
                src="https://avatars3.githubusercontent.com/u/6791502?v=3&s=200"
              />
            </span>
            <span
              className="article-subheading-value"
            >
              <strong />
            </span>
          </a>
        </span>
        <span>
          <span
            className="article-card-subheading-key"
          >
            tags:
          </span>
          <a
            href="/topics/football/articles"
            onClick={[Function]}
          >
            <span
              className="article-card-subheading-value"
            >
              <strong>
                football
              </strong>
            </span>
          </a>
        </span>
        <span>
          <span
            className="article-card-subheading-key"
          >
            comments:
          </span>
          <a
            href="#comments"
          >
            <span
              className="article-card-subheading-value"
            >
              <strong>
                0
              </strong>
            </span>
          </a>
        </span>
      </p>
      <div
        className="article-body"
      >
        something
      </div>
    </div>
  </div>
</div>
`;
